pubnub/pnconfiguration.py

```diff
@@ -36,7 +36,7 @@ def __init__(self):
         self._heartbeat_interval = PNConfiguration.DEFAULT_HEARTBEAT_INTERVAL
 
     def validate(self):
-        assert self.uuid is None or isinstance(self.uuid, str)
+        assert self.uuid and isinstance(self.uuid, str), "UUID missing or invalid type"
```

Comment:
Is this object immutable? Is it possible to change this `uuid` to something else after we're done validating?

Reply:
Let's leave it as it is. this change is already a small revolution. Validation of UUID is now triggered in the moment of setting it in `PNConfiguration` and during initialization of `PubNub` object